instrumentation/testing/benchmark.go

```diff
@@ -0,0 +1,197 @@
+package testing
+
+import (
+	"context"
+	stdErrors "errors"
+	"fmt"
+	"math"
+	"reflect"
+	"regexp"
+	"runtime"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+	"unsafe"
+
+	"github.com/opentracing/opentracing-go"
+
+	"go.undefinedlabs.com/scopeagent/ast"
+	"go.undefinedlabs.com/scopeagent/instrumentation"
+)
+
+type (
+	Benchmark struct {
+		b *testing.B
+	}
+)
+
+var (
+	instrumentedBenchmarkMutex sync.RWMutex
+	instrumentedBenchmark      = map[*testing.B]*Benchmark{}
```

Comment:
The test equivalent is called `autoInstrumentedTests`. Maybe we can follow the same convention? i.e. in both with or without the `auto` prefix. Same with the methods `StartBenchmark`, `isBenchmarkInstrumented`, etc.

Reply:
ok, I did some naming changes to follow conventions.